src/zarr/core/group.py

```diff
@@ -294,6 +294,36 @@ def flatten(
         return metadata
 
 
+class ArraysProxy:
+    """
+    Proxy for arrays in a group.
+
+    Used to implement the `Group.arrays` property
+    """
+
+    def __init__(self, group: Group) -> None:
+        self._group = group
+
+    def __getitem__(self, key: str) -> Array:
+        obj = self._group[key]
+        if isinstance(obj, Array):
+            return obj
+        raise KeyError(key)
+
+    def __setitem__(self, key: str, value: npt.ArrayLike) -> None:
+        """
+        Set an array in the group.
+        """
+        self._group._sync(self._group._async_group.set_array(key, value))
+
+    def __iter__(self) -> Generator[tuple[str, Array], None]:
+        for name, async_array in self._group._sync_iter(self._group._async_group.arrays()):
+            yield name, Array(async_array)
+
+    def __call__(self) -> Generator[tuple[str, Array], None]:
+        return iter(self)
```

Comment:
I could use some opinions here!

On one hand, `iter(group.arrays)` is nice and tidy. But it also doesn't match the existing API (`group.arrays()`) and there are other methods where symetry seems useful (`array_keys`, `array_values`, etc.). 

@d-v-b - you'll have an opinion here :)

Reply:
what breaks if we remove `__call__`? I feel like that's redundant if you can just iterate. And then we could add `keys` and `values` methods to the `ArraysProxy` object that lossily call `__iter__`, thereby replacing `array_keys` and `array_values`? 

We end up with a kind of a weird data structure -- something that's _asymptotically_ a dict? But I think that also matches the semantics of "lots of stuff stored on disk with unique names".

Reply:
> what breaks if we remove __call__ 

lots of stuff that looks like this (from our test suite):

```python
>       assert sorted(group.arrays(), key=lambda x: x[0]) == expected_arrays
E       TypeError: 'ArraysProxy' object is not callable
```

Looking at 2.18, I'm realizing we also are missing the `recurse` option to `arrays`... 

All of this makes me think it may be asking too much to overload `.arrays` with setiem/getitem/iter/etc.